### PR TITLE
[Merged by Bors] - feat(Topology/Order): continuity of `Finset.sup`, `partialSups` etc

### PR DESCRIPTION
**Pull request by urkud**
Also rename `Filter.Tendsto.sup_right_nhds` to `Filter.Tendsto.sup_nhds` etc.

---

- [ ] Should it be `Filter.Tendsto.nhds_sup`?
- [ ] Should I put lemmas about `Finset.sup` etc in a new file too?

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Order/PartialSups.lean [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Topology.Order.Lattice
+import Mathlib.Order.PartialSups
+
+/-!
+# Continuity of `partialSups`
+
+In this file we prove taht `partialSups` of a sequence of continuous functions is continuous
```
Comment (sgouezel) [inline Mathlib/Topology/Order/PartialSups.lean:12]:
```suggestion
In this file we prove that `partialSups` of a sequence of continuous functions is continuous
```

File: Mathlib/Topology/Order/Lattice.lean [hunk 2]
```diff
@@ -105,26 +104,270 @@ theorem Continuous.sup [Sup L] [ContinuousSup L] {f g : X → L} (hf : Continuou
...
+  simpa only [← Finset.sup'_apply] using finset_sup'_apply hne hs
+
+lemma ContinuousOn.finset_sup'_apply (hne : s.Nonempty) (hs : ∀ i ∈ s, ContinuousOn (f i) t) :
+    ContinuousOn (fun a ↦ s.sup' hne (f · a)) t := fun x hx ↦
+  ContinuousWithinAt.finset_sup'_apply hne fun i hi ↦ hs i hi x hx
+
+lemma ContinuousOn.finset_sup' (hne : s.Nonempty) (hs : ∀ i ∈ s, ContinuousOn (f i) t) :
+    ContinuousOn (s.sup' hne f) t := fun x hx ↦
+  ContinuousWithinAt.finset_sup' hne fun i hi ↦ hs i hi x hx
+
```
Comment (sgouezel) [inline Mathlib/Topology/Order/Lattice.lean:276]: Aren't you missing the `Continuous` version? Same thing several times below.

Comment (urkud) [reply Mathlib/Topology/Order/Lattice.lean:276]: We should have a command/attribute that generates `ContinuousAt`/`ContinuousWithinAt`/`ContinuousOn` from `Tendsto` (or from `ContinuousWithinAt`).

Comment (urkud) [reply Mathlib/Topology/Order/Lattice.lean:276]: Fixed. Thanks!

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**